src/integration/java/io/pinecone/integration/dataPlane/UpsertAndQueryServerlessTest.java

```diff
@@ -76,7 +76,7 @@ public void upsertOptionalVectorsAndQueryIndexSyncTest() throws Exception {
         index.upsert(vectors, namespace);
 
         // wait sometime for the vectors to be upserted
-        Thread.sleep(7500);
+        Thread.sleep(120000);
```

Comment:
Allowing more time to upsert

Reply:
This is a long time to wait for as many times as we're calling it, but if it gets tests passing for now that's fine. We should look at removing these and replacing with more granular polling rather than forcing a 2 minute wait we may not actually need.

Reply:
Yes, ideally remove delay or polling functionality completely in future